homeassistant/components/camera/__init__.py

```diff
@@ -1092,6 +1094,38 @@ async def websocket_update_prefs(
         connection.send_result(msg["id"], entity_prefs)
 
 
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "camera/snapshot",
+        vol.Required("entity_id"): cv.entity_id,
+    }
+)
+@websocket_api.async_response
+async def websocket_get_snapshot(
+    hass: HomeAssistant, connection: ActiveConnection, msg: dict[str, Any]
+) -> None:
+    """Handle get camera snapshot websocket command."""
+    try:
+        entity_id = msg["entity_id"]
+        camera = get_camera_from_entity_id(hass, entity_id)
+
+        async with asyncio.timeout(CAMERA_IMAGE_TIMEOUT):
+            image: bytes | None = (
+                await _async_get_stream_image(camera, wait_for_next_keyframe=True)
+                if camera.use_stream_for_stills
+                else await camera.async_camera_image()
```

Comment:
Generally we don't send large payloads over the websocket especially since a large payload will cause everything behind it to back up

Reply:
Its usually better to do a separate `GET` request for something like this

Reply:
Completely forgot that we already have an end point for this purpose